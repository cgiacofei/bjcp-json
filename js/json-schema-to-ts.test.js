const convert = require('./json-schema-to-ts.js')
const rootSchema = require('../json/beer.json')

test('simple schema converted to ts', () => {
  expect(
    convert({
      definitions: {
        ObjType: {
          type: 'object',
          properties: {
            strProp: { type: 'string' },
            numberProp: { type: 'number' },
            intProp: { type: 'integer' }
          }
        }
      }
    })
  ).toBe(
    `    export type ObjType = {
        strProp?: string,
        numberProp?: number,
        intProp?: number,
    }

`
  )
})

test('enum format', () => {
  expect(
    convert({
      definitions: {
        EnumType: {
          type: 'object',
          properties: {
            enumProp: { type: 'string', enum: ['a', 'b'] }
          }
        }
      }
    })
  ).toMatchSnapshot()
})
