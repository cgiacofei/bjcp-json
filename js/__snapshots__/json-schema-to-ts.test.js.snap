// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum format 1`] = `
"    export type EnumType = {
        enumProp?:  | \\"a\\" | \\"b\\",
    }

"
`;
